lib/datadog/ci/context/local.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "datadog/core/utils/sequence"
+
+module Datadog
+  module CI
+    module Context
+      class Local
+        def initialize
+          @key = "datadog_ci_active_test_#{Local.next_instance_id}"
+        end
+
+        def activate_test!(test)
+          raise "Nested tests are not supported. Currently active test: #{active_test}" unless active_test.nil?
+
+          if block_given?
+            begin
+              self.active_test = test
+              yield
+            ensure
+              self.active_test = nil
+            end
+          else
+            self.active_test = test
+          end
+        end
+
+        def deactivate_test!(test)
+          return if active_test.nil?
+
+          if active_test == test
+            self.active_test = nil
+          else
+            raise "Trying to deactivate test #{test}, but currently active test is #{active_test}"
+          end
+        end
+
+        def active_test
+          Thread.current[@key]
+        end
+
+        UNIQUE_INSTANCE_MUTEX = Mutex.new
+        UNIQUE_INSTANCE_GENERATOR = Datadog::Core::Utils::Sequence.new
+
+        private_constant :UNIQUE_INSTANCE_MUTEX, :UNIQUE_INSTANCE_GENERATOR
+
+        def self.next_instance_id
+          UNIQUE_INSTANCE_MUTEX.synchronize { UNIQUE_INSTANCE_GENERATOR.next }
+        end
+
+        private
+
+        def active_test=(test)
+          Thread.current[@key] = test
+        end
```

Comment:
I wonder if we could get away with not having a unique id per context. Is it really that common that we'll have more than one context alive in an app?

I'm asking because this has the same potential memory leak issues I shared in our private chat; namely, that the unique keys will live forever until the Ruby process dies.

Reply:
it is very convenient for testing though, not sure how often would it be used in real world case

what is important for us is that CI session is short lived so there would be one recorder instance per forked process and there is no real danger in creating really bad memory leak in my opinion

Reply:
Note that I'm not saying to create a single context forever; what I'm saying is, if the semantics is that in the usual case only one context would be alive, it seems like you could reuse the key?

Reply:
yes, this is true, let me try to do that and see how many tests break...

Reply:
@ivoanjo it worked without issues and made the code simpler! thank you for the suggestion